tests/models/faraday/test_faraday_model.py

```diff
@@ -141,3 +145,43 @@ def test_filter_mask():
     assert torch.equal(got_kwh, expected_tensor)
     for feature in got_features:
         assert torch.equal(got_features[feature], expected_tensor)
+
+
+def test_prepare_gmm_input():
+
+    vae_module = FaradayVAE(
+        class_dim=2, latent_dim=16, learning_rate=0.001, mse_weight=3
+    )
+    batch = TrainingData(
+        kwh=torch.rand(100, 48),
+        features={"feature_1": torch.rand(100), "feature_2": torch.rand(100)},
+        weights=torch.randint(low=1, high=5, size=(100,)),
+    )
+    encoded_batch = prepare_data_for_model(vae_module, batch)
+
+    assert encoded_batch.shape[0] == batch["weights"].sum()
+    assert encoded_batch.shape[1] == 16 + 2
```

Comment:
Just adding some explanation of why this is 16+2. 
```suggestion
    assert encoded_batch.shape[1] == 16 + 2 # Latent dim + number of extra features
```

Alternatively we could make it

`assert encoded_batch.shape[1] == vae_module.latent_dim + len(batch["features"].keys())`